packages/flutter/lib/src/foundation/change_notifier.dart

```diff
@@ -210,7 +243,8 @@ class ChangeNotifier implements Listenable {
   @mustCallSuper
   void dispose() {
     assert(_debugAssertNotDisposed());
-    _listeners = null;
+    _listeners = List<VoidCallback?>.filled(0, null);
+    _disposed = true;
```

Comment:
You can actually just leave _listeners as is, there's no reason to clear or reset the list. The garbage collector will take care of it. The only reason we set it to null before was as a flag that it was disposed, but not _disposed does that.